x-pack/plugin/core/src/main/java/org/elasticsearch/license/RestPutLicenseAction.java

```diff
@@ -21,7 +21,7 @@
 
 public class RestPutLicenseAction extends BaseRestHandler {
 
-    RestPutLicenseAction() {}
+    public RestPutLicenseAction() {}
```

Comment:
for these actions that getting public constructors, can you move them 
```
package org.elasticsearch.license;
```
to
```
package  org.elasticsearch.license.internal;
```
We don't want 3rd party plugins to manage a license and as such we [conditionally export ](https://github.com/elastic/elasticsearch/blob/main/x-pack/plugin/core/src/main/java/module-info.java#L27) `org.elasticsearch.license.internal` such that a user can't accidentally use classes from the internal package.

Reply:
Turns out that's actually not so easy, I'd have to move a bunch of other classes as well or make them public in turn (e.g. the associated request + builder classes). Fine to move all of that to the `internal` package?

Reply:
Seems like exposing the rest actions doesn't really make anything worse necessarily? We already have both the `PutLicenseAction` and the request class public so someone could just exploit that already?

Reply:
Ah... in that case, don't worry about it. It's just a way to help express intent (not a security thing).